pkg/kubelet/kubelet_node_status.go

```diff
@@ -273,6 +273,23 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 			glog.Infof("Adding node label from cloud provider: %s=%s", metav1.LabelInstanceType, instanceType)
 			node.ObjectMeta.Labels[metav1.LabelInstanceType] = instanceType
 		}
+
+		labels, err := instances.NodeLabels(kl.nodeName)
+		if err != nil {
+			return nil, err
+		}
+		if labels != nil {
+			prefix = "cloud.alpha.kubernetes.io/"
+			for name, value := range labels {
+				key := prefix + name
+				if len(key) > 63 {
+					glog.Warningf("Key: %s is too long, skipping", key)
```

Comment:
+1. The lack of reconciliation could make the labels dangerous for cluster admins to rely on. Even if you restarting kubelet, it won't update the labels based on the new tags. We've already seen quite a lot of confusion about node labels from users. I think it might be better to look for other alternatives (e.g., a separate controller or a script which reconciles periodically). At the least, this should be feature gated and marked as alpha.